### PR TITLE
Tokendriver

### PR DESCRIPTION
**Pull request by heckj**
termie - 

is this acceptable for what you had in mind as an abstract base class defining the token service?

### CODE REVIEW

**Comment by termie** [comment]
Looks great, just add newline and period to last doc string, and remove those merge commits?

**Comment by heckj** [comment]
done - and I also changed the params internal to the docstrings. Is this format acceptable?

File: keystone/token/core.py [hunk 1]
```diff
@@ -19,3 +19,50 @@ class Manager(manager.Manager):
...
+
+        :param token_id: identity of the token
+        :type token_id: string
+        :returns: token_ref or None.
+
+        """
+        raise NotImplementedError()
+
+    def create_token(self, token_id, data):
+        """create a token by id and data.
```
Comment (termie) [inline keystone/token/core.py]: capital C pls :p (sorry)

**Comment by heckj** [comment]
added period and compressed the changes into a single update

**Comment by termie** [comment]
groovy, for the record i think most of the time the "type" addition to the docs is not necessary unless it is unusual, but no harm in adding it

**Pull request merged**